solr4/src/main/scala/io/phdata/pulse/solr/SolrProvider.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 phData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.phdata.pulse.solr
```

Comment:
These solrprovider classes know how to instantiate a SolrService. They don't have an interface but are compatible between the solr4 and solr7 projects